### PR TITLE
change(tests): add acceptance test for getblocktemplate method in CI

### LINKED ISSUE #5652: Add an acceptance test for the getblocktemplate method
## Motivation

The vector and snapshot tests for the `getblocktemplate` RPC method won't be able to use the non-finalized state to test the contextual validity checks for https://github.com/ZcashFoundation/zebra/pull/5630.

## Designs

- Use `from_script_hash(Mainnet, [0x7e; 20])` as the `miner_address` in the config
- Sync Zebra to the network tip
- Call the getblocktemplate RPC method
- Assert that the call was successful

### PR DESCRIPTION
**Pull request by arya2**
## Motivation

Needed for CI to test the `getblocktemplate` method with the sync to tip and contextual validation checks.

Closes #5652

## Solution

- Add a `miner_address` to the config when the `getblocktemplate-rpcs` feature is enabled
- Sync to the tip
- Call getblocktemplate RPC method and assert that the request was successful
- Add new test to CI

## Review

Anyone can review.

### Reviewer Checklist

  - [x] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [x] Are the PR labels correct?
  - [x] Does the code do what the ticket and PR says?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

File: zebrad/tests/acceptance.rs [hunk 1]
```diff
@@ -103,6 +103,12 @@
 //!
 //! ## Getblocktemplate tests
 //!
+//! Example of how to run the get_block_template test:
+//!
+//! ```console
+//! ZEBRA_CACHED_STATE_DIR=/path/to/zebra/chain cargo test get_block_template --features getblocktemplate-rpcs --release  -- --ignored --nocapture
+//! ```
```
Comment (dconnolly) [inline zebrad/tests/acceptance.rs:110]: 💖

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -78,6 +78,10 @@ case "$1" in
             ls -lh "$ZEBRA_CACHED_STATE_DIR"/*/* || (echo "No $ZEBRA_CACHED_STATE_DIR/*/*"; ls -lhR  "$ZEBRA_CACHED_STATE_DIR" | head -50 || echo "No $ZEBRA_CACHED_STATE_DIR directory")
             ls -lhR "$LIGHTWALLETD_DATA_DIR/db" || (echo "No $LIGHTWALLETD_DATA_DIR/db"; ls -lhR "$LIGHTWALLETD_DATA_DIR" | head -50 || echo "No $LIGHTWALLETD_DATA_DIR directory")
             cargo test --locked --release --features lightwalletd-grpc-tests --package zebrad --test acceptance -- --nocapture --include-ignored sending_transactions_using_lightwalletd
+        elif [[ "$TEST_GET_BLOCK_TEMPLATE" -eq "1" ]]; then
+            # Starting with a cached Zebra tip, test getting a block template from Zebra's RPC server.
+            ls -lh "$ZEBRA_CACHED_STATE_DIR"/*/* || (echo "No $ZEBRA_CACHED_STATE_DIR/*/*"; ls -lhR  "$ZEBRA_CACHED_STATE_DIR" | head -50 || echo "No $ZEBRA_CACHED_STATE_DIR directory")
+            cargo test --locked --release --features getblocktemplate-rpcs --package zebrad --test acceptance -- --nocapture --include-ignored get_block_template
```
Comment (dconnolly) [inline docker/entrypoint.sh:84]: 👍

**Review by dconnolly — APPROVED** [review]
lgtm!

**Comment by mpguerra** [comment]
@mergifyio update

**Pull request merged**